sdk/keyvault/keyvault-certificates/samples/mergeCertificate.ts

```diff
@@ -0,0 +1,59 @@
+import { CertificatesClient } from "../src";
+import { DefaultAzureCredential } from "@azure/identity";
+
+// This sample creates a certificate with an Unknown issuer, then signs this certificate using a fake
+// certificate authority and the mergeCertificate API method.
+
+async function main(): Promise<void> {
+  // If you're using MSI, DefaultAzureCredential should "just work".
+  // Otherwise, DefaultAzureCredential expects the following three environment variables:
+  // - AZURE_TENANT_ID: The tenant ID in Azure Active Directory
+  // - AZURE_CLIENT_ID: The application (client) ID registered in the AAD tenant
+  // - AZURE_CLIENT_SECRET: The client secret for the registered application
+  const vaultName = process.env["KEYVAULT_NAME"] || "<keyvault-name>";
+  const url = `https://${vaultName}.vault.azure.net`;
+  const credential = new DefaultAzureCredential();
+
+  const client = new CertificatesClient(url, credential);
+
+  // Creating a certificate with an Unknown issuer.
+  await client.createCertificate("MyCertificate2", {
+    issuerParameters: {
+      name: "Unknown",
+      certificateTransparency: false
+    },
+    x509CertificateProperties: { subject: "cn=MyCert" }
+  });
+
+  // Retrieving the certificate request
+  const { csr } = await client.getCertificateOperation(certificateName);
+  const base64Csr = Buffer.from(csr!).toString("base64");
+  const wrappedCsr = `-----BEGIN CERTIFICATE REQUEST-----
+${base64Csr}
+-----END CERTIFICATE REQUEST-----`;
+  fs.writeFileSync("test.csr", wrappedCsr);
+
+  // Now, signing the retrieved certificate request with a fake certificate authority.
+  // A certificate authority is composed of two pieces, a certificate and a private key.
+  // We made these using openssl, as follows:
+  //   openssl genrsa -out ca.key 2048
+  //   openssl req -new -x509 -key ca.key -out ca.crt
+  childProcess.execSync(
+    "openssl x509 -req -in test.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out test.crt"
+  );
```

Comment:
I wouldn't do this here, just document what the user should do and instead ship a pre-made example file they could use

Reply:
There are two challenges here:
1. The certificate is being created using the API, this means that the `csr` is unique every time this sample is executed, which means that if we store a pre-generated file (for test.csr or test.crt), this test will fail.
2. Users can't simply stop the sample in the middle of its tracks and run the sign command that you're pointing at.

Since this is part of OpenSSL, some languages do have a way to make these calls inside of the language, without using something like child_process. Node, however, has a very limited crypto package that doesn't seem to have an API for certificate requests and x509 operations.

Let me know what you think and what direction you recommend!

Reply:
I was thinking we could just provide a certificate we made as part of the samples, so they don't have to go messing with openssl to try it out.

Reply:
The issue is that mergeCertificate can only work on a pending certificate, so both calls, createCertificate and mergeCertificate, have to happen. The middle work, which is to sign the certificate request, is what can't be done with our API, nor NodeJS's crypto, but only OpenSSL.

Let's say I create a certificate locally and sign it. Fine, but that can't work with mergeCertificate, since there's no pending certificate remotely for what we have locally.

I could be missing something though!

Reply:
I'll update the sample with documentation on how to install openssl on Windows, Linux and Mac.

Reply:
Ah okay, got it! Then as long as we have instructions for our three platforms about how to set up OpenSSL, I'm good.

Reply:
Done!